databroker/src/broker.rs

```diff
@@ -1561,6 +1623,290 @@ impl<'a, 'b> AuthorizedAccess<'a, 'b> {
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    pub async fn actuate(
+        &self,
+        vss_id: &i32,
+        data_value: &DataValue,
+    ) -> Result<(), (ActuationError, String)> {
+        let vss_id = *vss_id;
+        let result_entry = self.get_entry_by_id(vss_id).await;
+        match result_entry {
+            Ok(entry) => {
+                let vss_path = entry.metadata.path;
+                let result_can_write_actuator =
+                    self.permissions.can_write_actuator_target(&vss_path);
+                match result_can_write_actuator {
+                    Ok(_) => {}
+                    Err(PermissionError::Denied) => {
+                        let message = format!("Permission denied for vss_path {}", vss_path);
+                        return Err((ActuationError::PermissionDenied, message));
+                    }
+                    Err(PermissionError::Expired) => {
+                        return Err((
+                            ActuationError::PermissionExpired,
+                            "Permission expired".to_string(),
+                        ))
+                    }
+                }
+            }
+            Err(ReadError::NotFound) => {
+                let message = format!("Could not resolve vss_path of vss_id {}", vss_id);
+                return Err((ActuationError::NotFound, message));
+            }
+            Err(ReadError::PermissionDenied) => {
+                let message = format!("Permission denied for vss_id {}", vss_id);
+                return Err((ActuationError::PermissionDenied, message));
+            }
+            Err(ReadError::PermissionExpired) => {
+                return Err((
+                    ActuationError::PermissionExpired,
+                    "Permission expired".to_string(),
+                ))
+            }
+        }
+
+        let entry_update = EntryUpdate {
+            path: None,
+            datapoint: None,
+            actuator_target: Some(Some(Datapoint {
+                ts: SystemTime::now(),
+                source_ts: None,
+                value: data_value.clone(),
+            })),
+            entry_type: None,
+            data_type: None,
+            description: None,
+            allowed: None,
+            unit: None,
+        };
+
+        let entry_updates = [(vss_id, entry_update)];
+        let legacy_result = self.update_entries(entry_updates).await;
+        if legacy_result.is_err() {
+            let update_errors = legacy_result.unwrap_err();
+            let opt_error = update_errors.first();
+            if opt_error.is_some() {
+                let error = opt_error.unwrap();
+                let message = format!(
+                    "Could not set actuator target for vss_id {}: {:?}",
+                    vss_id, error.1
+                );
+                warn!(message);
+            }
+        }
+
+        let mut guard = self.broker.subscriptions.write().await;
```

Comment:
you just need read acces here right? Like in 1731

Reply:
I execute cleanup in line 1890 after checking if any actuation provider expired, so I think right now it's still feasible. However it's still challengable if I should call cleanup here, or we maybe don't need to call it at all.

Reply:
While trying to introduce the cleanup operation in batch_actuate it caused a panic during `let legacy_result = self.update_entries(entry_updates).await;`. The panic must somehow be caused by the write guard, be. So I removed the cleanup from both methods again and converted the guard back to a read guard. 

As the housekeeping task is executed every second, it should be sufficient to not require manually triggering the cleanup. 

However I also added the expiration check, which was missing from batch_actuate.